subscriber.rb

```diff
@@ -0,0 +1,48 @@
+
+require 'aws-sdk-sqs'
+require 'byebug'
+require_relative 'load_aws'
+
+class Subscriber
+  
+  ACCESS_KEY_ID = AWS["access_key_id"]
+  SECRET_ACCESS_KEY = AWS["secret_access_key"]
+  REGION = AWS["region"]
+
+  attr_reader :sqs
+
+  def initialize
+    creds = Aws::Credentials.new(ACCESS_KEY_ID, SECRET_ACCESS_KEY)
+    @sqs = Aws::SQS::Client.new(region: REGION, credentials: creds)
+  end
+
+# Todo: URL validation
+  def read_queue(options=[])
+    queue_url = options[0]
+    max_number_of_messages = options[1].nil? ? 10 : options[1]
```

Comment:
Don't give an option. Maximum 1 message can be read from the queue at any time.

Reply:
If we do this, it will fetch at max one message in each execution and halt and we will have to run the execution command  again and again for each message. Polling and receive_message work differently.

Reply:
You could do an infinite loop for subscriber codel. Have the subscriber being called in the infinite loop with a sleep in between subscriber requests.
```
loop do
  call_subscriber_service
  sleep 5 seconds
end
```

Reply:
We could do this. But its not a good practice to use it this way. We could put it in an infinite loop but fetch more than one message at a time. Let me know what you think. For now, I have made the changes you suggested but not entirely convinced of it being the best way to receive messages.